sql/core/src/test/scala/org/apache/spark/sql/execution/python/PythonUDFSuite.scala

```diff
@@ -86,6 +86,61 @@ class PythonUDFSuite extends QueryTest with SharedSparkSession {
     checkAnswer(actual, expected)
   }
 
+  test("variant input to pandas grouped agg UDF") {
+    assume(shouldTestPandasUDFs)
+    val df = spark.range(0, 10).selectExpr(
+      """parse_json(format_string('{"%s": "test"}', id)) as v""")
+
+    val testUdf = TestGroupedAggPandasUDFStringifiedMax(name = "pandas_udf")
+    checkError(
+      exception = intercept[AnalysisException] {
+        df.agg(testUdf(df("v"))).collect()
+      },
+      errorClass = "DATATYPE_MISMATCH.UNSUPPORTED_UDF_INPUT_TYPE",
+      parameters = Map("sqlExpr" -> "\"pandas_udf(v)\"", "dataType" -> "VARIANT"))
+  }
+
+  test("complex variant input to pandas grouped agg UDF") {
```

Comment:
To @richardc-db, @allisonwang-db , @HyukjinKwon .

This test case seems to miss `assume(shouldTestPandasUDFs)` mistakenly.